src/profile.jl

```diff
@@ -0,0 +1,170 @@
+struct FeProfile{T<:AbstractFloat}  # derived model with the j'th fixed-effects coefficient held constant
+    m::LinearMixedModel{T}          # copy of original model after removing the j'th column from X
+    y₀::Vector{T}                   # original response vector
+    xⱼ::Vector{T}                   # the column that was removed from X
+    j::Integer
+end
+
+"""
+    Base.copy(ReMat{T,S})
+
+Return a shallow copy of ReMat.
+
+A shallow copy shares as much internal storage as possible with the original ReMat.
+Only the vector `λ` and the `scratch` matrix are copied.
+"""
+function Base.copy(ret::ReMat{T,S}) where {T,S}
+    return ReMat{T,S}(ret.trm,
+        ret.refs,
+        ret.levels,
+        ret.cnames,
+        ret.z,
+        ret.wtz,
+        copy(ret.λ),
+        ret.inds,
+        ret.adjA,
+        copy(ret.scratch))
+end
+
+function FeProfile(m::LinearMixedModel, j::Integer)
+    Xy = m.Xymat.xy
+    xcols = collect(axes(Xy, 2))
+    ycol = pop!(xcols)
+    notj = deleteat!(xcols, j) # indirectly check that j ∈ xcols
+    y₀ = Xy[:, ycol]
+    xⱼ = Xy[:, j]
+    feterm = FeTerm(Xy[:, notj], m.feterm.cnames[notj])
+    reterms = [copy(ret) for ret in m.reterms]
+    m = fit!(LinearMixedModel(y₀ - xⱼ * m.β[j], feterm, reterms, m.formula); progress=false)
+    @. m.optsum.initial = max(m.optsum.initial, m.lowerbd + 0.05)
+    return FeProfile(m, y₀, xⱼ, j)
+end
+
+function betaprofile!(pr::FeProfile{T}, val, βⱼ::T, j::Integer, obj::T) where {T}
+    prm = pr.m
+    refit!(prm, mul!(copyto!(pr.m.y, pr.y₀), pr.xⱼ, βⱼ, -1, 1); progress=false)
+    ζ = sqrt(prm.objective - obj)
+    push!(val.ζ, ζ)
+    push!(val.θ, prm.θ)
+    push!(val.σ, prm.σ)
+    β = prm.β
+    splice!(β, j:(j - 1), βⱼ)
+    push!(val.β, β)
+    return ζ
+end
+
+function profileβj(m::LinearMixedModel{T}, j::Integer; threshold=4) where {T}
+    @compat (; β, θ, σ, stderror, objective) = m
+    val = (; ζ = -T[0], β = [SVector{length(β)}(β)], σ = [σ], θ = [SVector{length(θ)}(θ)])
+    prj = FeProfile(m, j)
+    st = stderror[j] * 0.5
+    bb = β[j] - st
+    while true
+        betaprofile!(prj, val, bb, j, objective) < threshold || break
+        bb -= st
+    end
+    rmul!(val.ζ, -1)
+    bb = β[j] + st
+    while true
+        betaprofile!(prj, val, bb, j, objective) < threshold || break
+        bb += st
+    end
+    return sort!(Table(merge((; par = fill(Symbol("β$j"), length(val.σ))), val)), by = r -> r.ζ)
+end
+
+struct MixedModelProfile
+    tbl::Table                       # Table containing ζ, σ, β, and θ from each conditional fit
+    fecnames::Vector{String}         # Fixed-effects coefficient names
+    facnames::Vector{Symbol}         # Names of grouping factors
```

Comment:
the other two names are vectors (of vectors) of strings

Reply:
Since this is for us, let's keep it easy.

But we should give this a second thought for consistency with `coefnames` etc. when it comes time for the next breaking release.